src/octoprint/plugins/gcodeviewer/__init__.py

```diff
@@ -70,6 +73,19 @@ def on_settings_migrate(self, target, current):
                     self._settings.set_int(["sizeThreshold"], config["sizeThreshold"])
                 self._settings.global_remove(["gcodeViewer"])
 
+    def on_api_get(self, request):
```

Comment:
The way this is implemented heavily deviates from other file related endpoints. Initially I thought it would be easiest to just have the get point only operate on the currently selected file, making the `SimpleApiPlugin` the easiest approach to implementation since no parameter transfer would be needed, but if you want to go with a parameter based approach (which also has its valid reasons, to combat issues caused by race conditions should the user go nuts with selecting new files), this should rather be a `BlueprintPlugin` with a `GET` method and the file path encoded into its path.

That has the upside that we can also just return 404 if we can't find the file, instead of claiming "nope, clean".

Something like this (assuming the mixin `octoprint.plugin.BlueprintPlugin` mixin is also included above):

``` python
@octoprint.plugin.BlueprintPlugin.route("/skipuntilcheck/<string:origin>/<path:filename>", methods=["GET"])
def check_skip_until_presence(self, origin, path):
    try:
        path = self._file_manager.path_on_disk(origin, path)
    except NotImplementedError:
        # storage doesn't support path on disk
        flask.abort(404)

    if not os.path.exists(path):
        # path doesn't exist
        flask.abort(404)

    skipUntilThis = self._settings.get(["skipUntilThis"])
    if not skipUntilThis:
        # no skipUntilThis, no need to search, shortcut
        return flask.jsonify(present=False)

    return flask.jsonify(present=search_through_file(path, skipUntilThis))

def is_blueprint_csrf_protected(self):
    return True
```

Reply:
I see you return True on the csrf protected method. Do I need to do something in the worker for that?

Reply:
No, not since you are only using a `GET` endpoint. If at some point you'd want to `POST`, `PUT`, `PATCH`, `DELETE` stuff, the `fetch` calls would need some love to add some headers. The `return True` here is merely so that an explicit decision has been made for this plugin, because otherwise OctoPrint will implicitly make it (`False`) and log two warnings, one for the implicit decision, one for the disabled CSRF protection. See the heads-ups in the release notes of 1.8.3.